### PR TITLE
[CustomerAccountsExtensibility] Add navigation api

### PR DESCRIPTION
**Pull request by oluwatimio**
### Background

Needed for https://github.com/Shopify/customer-account-web/pull/2677

### Solution

Please tophat https://github.com/Shopify/customer-account-web/pull/2677

### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

File: packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  type: NavigationType;
+  from: NavigationHistoryEntry;
+}
+
+export interface StandardExtensionNavigation {
+  navigate: NavigateFunction;
+}
+
+export type FullExtensionNavigation = StandardExtensionNavigation & {
+  currentEntry: NavigationHistoryEntry;
```
Comment (oluwatimio) [inline packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts:24]: Will add the listeners in a follow up pr when Implemented in customer-account-web

**Review by brianshen1990 — APPROVED** [review]
Make sure `NavigationHistoryEntry.state` is a intended change, others looks great

File: packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+export interface NavigationOptions {
+  state?: Record<string, any>;
+  type: 'push' | 'replace';
+}
+
+export interface NavigationHistoryEntry {
+  key: string;
+  url: string;
+  state: Record<string, any>;
```
[the remark this answers is not in the document]
Comment (robin-drexler) [reply packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts:11]: I think it should be a function if that's in the spec, just to stay consistent.

File: packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+export type NavigationType = 'push' | 'replace' | 'traverse';
+
+export interface NavigationOptions {
+  state?: Record<string, any>;
+  type: 'push' | 'replace';
```
Comment (robin-drexler) [inline packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts:5]: shouldn't that be `history`?  https://developer.mozilla.org/en-US/docs/Web/API/Navigation/navigate#:~:text=as%20info.-,history,-An%20enumerated%20value

https://github.com/Shopify/ui-api-design/discussions/21#discussioncomment-6071362

File: packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export interface NavigationHistoryEntry {
+  key: string;
+  url: string;
+  state: Record<string, any>;
+}
+
+export interface NavigationCurrentEntryChangeEvent {
+  type: NavigationType;
```
Comment (robin-drexler) [inline packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts:15]:
```suggestion
  navigationType: NavigationType;
```

https://developer.mozilla.org/en-US/docs/Web/API/NavigationCurrentEntryChangeEvent#:~:text=was%20navigated%20from.-,navigationType,-Read%20only

**Review by robin-drexler — APPROVED** [review]

**Pull request merged**